// react-bootstrap-toasts/dist/index.mjs
import { ToastContainer as fr, Toast as ee } from "react-bootstrap";
import Se, { createContext as dr, forwardRef as vr, useState as xe, useImperativeHandle as pr, useRef as hr, useContext as gr, useMemo as yr, useEffect as mr } from "react";
const je = dr(void 0);
var re = { exports: {} }, Y = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Pe;
function br() {
  if (Pe)
    return Y;
  Pe = 1;
  var m = Se, w = Symbol.for("react.element"), C = Symbol.for("react.fragment"), b = Object.prototype.hasOwnProperty, f = m.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, v = { key: !0, ref: !0, __self: !0, __source: !0 };
  function R(p, d, O) {
    var h, _ = {}, g = null, S = null;
    O !== void 0 && (g = "" + O), d.key !== void 0 && (g = "" + d.key), d.ref !== void 0 && (S = d.ref);
    for (h in d)
      b.call(d, h) && !v.hasOwnProperty(h) && (_[h] = d[h]);
    if (p && p.defaultProps)
      for (h in d = p.defaultProps, d)
        _[h] === void 0 && (_[h] = d[h]);
    return { $$typeof: w, type: p, key: g, ref: S, props: _, _owner: f.current };
  }
  return Y.Fragment = C, Y.jsx = R, Y.jsxs = R, Y;
}
var L = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Oe;
function Er() {
  return Oe || (Oe = 1, process.env.NODE_ENV !== "production" && function() {
    var m = Se, w = Symbol.for("react.element"), C = Symbol.for("react.portal"), b = Symbol.for("react.fragment"), f = Symbol.for("react.strict_mode"), v = Symbol.for("react.profiler"), R = Symbol.for("react.provider"), p = Symbol.for("react.context"), d = Symbol.for("react.forward_ref"), O = Symbol.for("react.suspense"), h = Symbol.for("react.suspense_list"), _ = Symbol.for("react.memo"), g = Symbol.for("react.lazy"), S = Symbol.for("react.offscreen"), T = Symbol.iterator, u = "@@iterator";
    function j(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = T && e[T] || e[u];
      return typeof r == "function" ? r : null;
    }
    var x = m.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function y(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), n = 1; n < r; n++)
          t[n - 1] = arguments[n];
        De("error", e, t);
      }
    }
    function De(e, r, t) {
      {
        var n = x.ReactDebugCurrentFrame, i = n.getStackAddendum();
        i !== "" && (r += "%s", t = t.concat([i]));
        var s = t.map(function(o) {
          return String(o);
        });
        s.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, s);
      }
    }
    var Fe = !1, Ae = !1, Ie = !1, We = !1, $e = !1, te;
    te = Symbol.for("react.module.reference");
    function Ye(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === b || e === v || $e || e === f || e === O || e === h || We || e === S || Fe || Ae || Ie || typeof e == "object" && e !== null && (e.$$typeof === g || e.$$typeof === _ || e.$$typeof === R || e.$$typeof === p || e.$$typeof === d || e.$$typeof === te || e.getModuleId !== void 0));
    }
    function Le(e, r, t) {
      var n = e.displayName;
      if (n)
        return n;
      var i = r.displayName || r.name || "";
      return i !== "" ? t + "(" + i + ")" : t;
    }
    function ne(e) {
      return e.displayName || "Context";
    }
    function k(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && y("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case b:
          return "Fragment";
        case C:
          return "Portal";
        case v:
          return "Profiler";
        case f:
          return "StrictMode";
        case O:
          return "Suspense";
        case h:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case p:
            var r = e;
            return ne(r) + ".Consumer";
          case R:
            var t = e;
            return ne(t._context) + ".Provider";
          case d:
            return Le(e, e.render, "ForwardRef");
          case _:
            var n = e.displayName || null;
            return n !== null ? n : k(e.type) || "Memo";
          case g: {
            var i = e, s = i._payload, o = i._init;
            try {
              return k(o(s));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var F = Object.assign, W = 0, ae, oe, ie, se, ue, ce, le;
    function fe() {
    }
    fe.__reactDisabledLog = !0;
    function Ne() {
      {
        if (W === 0) {
          ae = console.log, oe = console.info, ie = console.warn, se = console.error, ue = console.group, ce = console.groupCollapsed, le = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: fe,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        W++;
      }
    }
    function Me() {
      {
        if (W--, W === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: F({}, e, {
              value: ae
            }),
            info: F({}, e, {
              value: oe
            }),
            warn: F({}, e, {
              value: ie
            }),
            error: F({}, e, {
              value: se
            }),
            group: F({}, e, {
              value: ue
            }),
            groupCollapsed: F({}, e, {
              value: ce
            }),
            groupEnd: F({}, e, {
              value: le
            })
          });
        }
        W < 0 && y("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var J = x.ReactCurrentDispatcher, G;
    function N(e, r, t) {
      {
        if (G === void 0)
          try {
            throw Error();
          } catch (i) {
            var n = i.stack.trim().match(/\n( *(at )?)/);
            G = n && n[1] || "";
          }
        return `
` + G + e;
      }
    }
    var H = !1, M;
    {
      var Ve = typeof WeakMap == "function" ? WeakMap : Map;
      M = new Ve();
    }
    function de(e, r) {
      if (!e || H)
        return "";
      {
        var t = M.get(e);
        if (t !== void 0)
          return t;
      }
      var n;
      H = !0;
      var i = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var s;
      s = J.current, J.current = null, Ne();
      try {
        if (r) {
          var o = function() {
            throw Error();
          };
          if (Object.defineProperty(o.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(o, []);
            } catch (D) {
              n = D;
            }
            Reflect.construct(e, [], o);
          } else {
            try {
              o.call();
            } catch (D) {
              n = D;
            }
            e.call(o.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (D) {
            n = D;
          }
          e();
        }
      } catch (D) {
        if (D && n && typeof D.stack == "string") {
          for (var a = D.stack.split(`
`), E = n.stack.split(`
`), c = a.length - 1, l = E.length - 1; c >= 1 && l >= 0 && a[c] !== E[l]; )
            l--;
          for (; c >= 1 && l >= 0; c--, l--)
            if (a[c] !== E[l]) {
              if (c !== 1 || l !== 1)
                do
                  if (c--, l--, l < 0 || a[c] !== E[l]) {
                    var P = `
` + a[c].replace(" at new ", " at ");
                    return e.displayName && P.includes("<anonymous>") && (P = P.replace("<anonymous>", e.displayName)), typeof e == "function" && M.set(e, P), P;
                  }
                while (c >= 1 && l >= 0);
              break;
            }
        }
      } finally {
        H = !1, J.current = s, Me(), Error.prepareStackTrace = i;
      }
      var I = e ? e.displayName || e.name : "", Ce = I ? N(I) : "";
      return typeof e == "function" && M.set(e, Ce), Ce;
    }
    function Ue(e, r, t) {
      return de(e, !1);
    }
    function Be(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function V(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return de(e, Be(e));
      if (typeof e == "string")
        return N(e);
      switch (e) {
        case O:
          return N("Suspense");
        case h:
          return N("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case d:
            return Ue(e.render);
          case _:
            return V(e.type, r, t);
          case g: {
            var n = e, i = n._payload, s = n._init;
            try {
              return V(s(i), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var U = Object.prototype.hasOwnProperty, ve = {}, pe = x.ReactDebugCurrentFrame;
    function B(e) {
      if (e) {
        var r = e._owner, t = V(e.type, e._source, r ? r.type : null);
        pe.setExtraStackFrame(t);
      } else
        pe.setExtraStackFrame(null);
    }
    function qe(e, r, t, n, i) {
      {
        var s = Function.call.bind(U);
        for (var o in e)
          if (s(e, o)) {
            var a = void 0;
            try {
              if (typeof e[o] != "function") {
                var E = Error((n || "React class") + ": " + t + " type `" + o + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[o] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw E.name = "Invariant Violation", E;
              }
              a = e[o](r, o, n, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (c) {
              a = c;
            }
            a && !(a instanceof Error) && (B(i), y("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", n || "React class", t, o, typeof a), B(null)), a instanceof Error && !(a.message in ve) && (ve[a.message] = !0, B(i), y("Failed %s type: %s", t, a.message), B(null));
          }
      }
    }
    var Je = Array.isArray;
    function z(e) {
      return Je(e);
    }
    function Ge(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function He(e) {
      try {
        return he(e), !1;
      } catch {
        return !0;
      }
    }
    function he(e) {
      return "" + e;
    }
    function ge(e) {
      if (He(e))
        return y("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", Ge(e)), he(e);
    }
    var $ = x.ReactCurrentOwner, ze = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, ye, me, K;
    K = {};
    function Ke(e) {
      if (U.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function Xe(e) {
      if (U.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function Qe(e, r) {
      if (typeof e.ref == "string" && $.current && r && $.current.stateNode !== r) {
        var t = k($.current.type);
        K[t] || (y('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', k($.current.type), e.ref), K[t] = !0);
      }
    }
    function Ze(e, r) {
      {
        var t = function() {
          ye || (ye = !0, y("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function er(e, r) {
      {
        var t = function() {
          me || (me = !0, y("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var rr = function(e, r, t, n, i, s, o) {
      var a = {
        $$typeof: w,
        type: e,
        key: r,
        ref: t,
        props: o,
        _owner: s
      };
      return a._store = {}, Object.defineProperty(a._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(a, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: n
      }), Object.defineProperty(a, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: i
      }), Object.freeze && (Object.freeze(a.props), Object.freeze(a)), a;
    };
    function tr(e, r, t, n, i) {
      {
        var s, o = {}, a = null, E = null;
        t !== void 0 && (ge(t), a = "" + t), Xe(r) && (ge(r.key), a = "" + r.key), Ke(r) && (E = r.ref, Qe(r, i));
        for (s in r)
          U.call(r, s) && !ze.hasOwnProperty(s) && (o[s] = r[s]);
        if (e && e.defaultProps) {
          var c = e.defaultProps;
          for (s in c)
            o[s] === void 0 && (o[s] = c[s]);
        }
        if (a || E) {
          var l = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          a && Ze(o, l), E && er(o, l);
        }
        return rr(e, a, E, i, n, $.current, o);
      }
    }
    var X = x.ReactCurrentOwner, be = x.ReactDebugCurrentFrame;
    function A(e) {
      if (e) {
        var r = e._owner, t = V(e.type, e._source, r ? r.type : null);
        be.setExtraStackFrame(t);
      } else
        be.setExtraStackFrame(null);
    }
    var Q;
    Q = !1;
    function Z(e) {
      return typeof e == "object" && e !== null && e.$$typeof === w;
    }
    function Ee() {
      {
        if (X.current) {
          var e = k(X.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function nr(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var Re = {};
    function ar(e) {
      {
        var r = Ee();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function _e(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = ar(r);
        if (Re[t])
          return;
        Re[t] = !0;
        var n = "";
        e && e._owner && e._owner !== X.current && (n = " It was passed a child from " + k(e._owner.type) + "."), A(e), y('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, n), A(null);
      }
    }
    function Te(e, r) {
      {
        if (typeof e != "object")
          return;
        if (z(e))
          for (var t = 0; t < e.length; t++) {
            var n = e[t];
            Z(n) && _e(n, r);
          }
        else if (Z(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var i = j(e);
          if (typeof i == "function" && i !== e.entries)
            for (var s = i.call(e), o; !(o = s.next()).done; )
              Z(o.value) && _e(o.value, r);
        }
      }
    }
    function or(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === d || r.$$typeof === _))
          t = r.propTypes;
        else
          return;
        if (t) {
          var n = k(r);
          qe(t, e.props, "prop", n, e);
        } else if (r.PropTypes !== void 0 && !Q) {
          Q = !0;
          var i = k(r);
          y("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", i || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && y("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function ir(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var n = r[t];
          if (n !== "children" && n !== "key") {
            A(e), y("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", n), A(null);
            break;
          }
        }
        e.ref !== null && (A(e), y("Invalid attribute `ref` supplied to `React.Fragment`."), A(null));
      }
    }
    function we(e, r, t, n, i, s) {
      {
        var o = Ye(e);
        if (!o) {
          var a = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (a += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var E = nr(i);
          E ? a += E : a += Ee();
          var c;
          e === null ? c = "null" : z(e) ? c = "array" : e !== void 0 && e.$$typeof === w ? (c = "<" + (k(e.type) || "Unknown") + " />", a = " Did you accidentally export a JSX literal instead of a component?") : c = typeof e, y("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", c, a);
        }
        var l = tr(e, r, t, i, s);
        if (l == null)
          return l;
        if (o) {
          var P = r.children;
          if (P !== void 0)
            if (n)
              if (z(P)) {
                for (var I = 0; I < P.length; I++)
                  Te(P[I], e);
                Object.freeze && Object.freeze(P);
              } else
                y("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Te(P, e);
        }
        return e === b ? ir(l) : or(l), l;
      }
    }
    function sr(e, r, t) {
      return we(e, r, t, !0);
    }
    function ur(e, r, t) {
      return we(e, r, t, !1);
    }
    var cr = ur, lr = sr;
    L.Fragment = b, L.jsx = cr, L.jsxs = lr;
  }()), L;
}
(function(m) {
  process.env.NODE_ENV === "production" ? m.exports = br() : m.exports = Er();
})(re);
const q = re.exports.jsx, ke = re.exports.jsxs, Rr = vr((m, w) => {
  const [C, b] = xe([]);
  pr(w, () => ({
    show: (v) => {
      b((R) => {
        const p = [...R];
        return p.push(v), m.limit && p.length > m.limit && p.shift(), p;
      });
    },
    hide: (v) => {
      b((R) => [...R].filter((p) => p.id !== v));
    }
  }));
  const {
    toastContainerProps: f
  } = m;
  return /* @__PURE__ */ q(fr, {
    ...f,
    children: C.map((v) => {
      const {
        headerContent: R,
        toastHeaderProps: p
      } = v, d = /* @__PURE__ */ q(ee.Header, {
        ...p,
        children: R
      }), {
        bodyContent: O,
        toastBodyProps: h
      } = v, _ = /* @__PURE__ */ q(ee.Body, {
        ...h,
        children: O
      }), {
        toastProps: g
      } = v, {
        onClose: S
      } = g != null ? g : {};
      return g == null || delete g.onClose, /* @__PURE__ */ ke(ee, {
        ...g,
        onClose: () => {
          b((T) => T.filter((u) => u.id !== v.id)), S == null || S();
        },
        children: [d, _]
      }, v.id);
    })
  });
}), Cr = ({
  children: m,
  toastContainerProps: w,
  limit: C
}) => {
  const b = hr(null);
  return /* @__PURE__ */ ke(je.Provider, {
    value: b,
    children: [m, /* @__PURE__ */ q(Rr, {
      ref: b,
      toastContainerProps: w,
      limit: C
    })]
  });
};
let _r = 0;
const Pr = () => {
  const [m, w] = xe([]), C = gr(je);
  if (C === void 0)
    throw Error(
      "`useToasts` must be used inside of a `ToastsProvider`, otherwise it will not function correctly."
    );
  const b = yr(() => {
    const f = (u) => {
      let j = _r++;
      return w((x) => [...x, { ...u, id: j }]), j;
    }, v = (u) => {
      var j;
      (j = C.current) == null || j.hide(u);
    }, R = (u) => f(T(u, "info")), p = (u) => f(T(u, "primary")), d = (u) => f(T(u, "secondary")), O = (u) => f(T(u, "success")), h = (u) => f(T(u, "danger")), _ = (u) => f(T(u, "warning")), g = (u) => f(T(u, "dark")), S = (u) => f(T(u, "light")), T = (u, j) => {
      const { toastProps: x } = u;
      let y = { ...x, bg: j };
      return {
        ...u,
        toastProps: y
      };
    };
    return {
      show: f,
      hide: v,
      info: R,
      primary: p,
      secondary: d,
      success: O,
      danger: h,
      warning: _,
      dark: g,
      light: S
    };
  }, [C.current]);
  return mr(() => {
    const { current: f } = C;
    f !== null && m.length && (m.forEach((v) => {
      f.show(v);
    }), w([]));
  }), b;
};
export {
  Cr as ToastsProvider,
  Pr as useToasts
};
